hack/make/run

```diff
@@ -56,5 +55,32 @@ args="--debug \
 	$extra_params"
 
 echo "${dockerd} ${args}"
-# shellcheck disable=SC2086
-exec "${dockerd}" ${args}
+if [ -n "$DELVE_PORT" ] && [ -z "$DOCKER_ROOTLESS" ]; then
+	# Run Docker daemon with debugger
+	echo "Exposing Delve debugger on port $DELVE_PORT"
+
+	# shellcheck disable=SC2086
+	exec dlv --listen=0.0.0.0:$DELVE_PORT \
```

Comment:
> 0.0.0.0

This really looks insecure

Reply:
Exposing things like debugger or daemon to 0.0.0.0 isn’t the most secure option indeed, usually it’s 127.0.0.1. However, this is the case with a non-containerized development environment - since we’re running stuff in the container here, we cannot expose it to 127.0.0.1, otherwise the processes from outside the container will not be able to reach the debugger. What’s more, in the same script the Docker daemon’s TCP socket is exposed in the same way:

```bash
args="--debug \
	--host "tcp://0.0.0.0:${listen_port}" --host "unix://${socket}" \
	--storage-driver "${DOCKER_GRAPHDRIVER}" \
	--userland-proxy="${DOCKER_USERLANDPROXY}" \
	$storage_params \
	$extra_params"
```

I’ve just noticed that the Docker port publishing is done slightly differently in the Makefile though:

```makefile
DOCKER_PORT_FORWARD := $(if $(DOCKER_PORT),-p "$(DOCKER_PORT)",)
DELVE_PORT_FORWARD := $(if $(DELVE_PORT),-p "$(DELVE_PORT):$(DELVE_PORT)",)
```

The `DOCKER_PORT_FORWARD` allows one to specify values like `127.0.0.1:2375:2375`, which would publish the port only to the host. I will modify the script and the Makefile to handle `DELVE_PORT` in the same way.

Reply:
Comments about the area and the internet is a great idea for the first time in years and